lib/static-server.ts

```diff
@@ -0,0 +1,437 @@
+				}
+
+				/* General byte range validation */
+				if (!isNaN(byteRange.from) && !isNaN(byteRange.to) && 0 <= byteRange.from && byteRange.from <= byteRange.to) {
+					byteRange.valid = true;
+				} else {
+					console.warn('Request contains invalid range header: ', splitRangeHeader);
+				}
+			} else {
+				console.warn('Request contains unsupported range header: ', rangeHeader);
+			}
+		}
+		return byteRange;
+	}
+
+	async respondNoGzip(
+		status: number, contentType: string, _headers: Headers, file: string, stat: fs.Stats,
+		req: http.IncomingMessage, res: http.ServerResponse
+	): Promise<Result> {
+		const mtime = Date.parse(stat.mtime as any);
+		const headers: Headers = {};
+		const clientETag = req.headers['if-none-match'];
+		const clientMTime = Date.parse(req.headers['if-modified-since']!);
+		const byteRange = this.parseByteRange(req, stat);
+		let startByte = 0;
+		let length = stat.size;
+
+		/* Handle byte ranges */
+		if (byteRange.valid) {
+			if (byteRange.to < length) {
+				// Note: HTTP Range param is inclusive
+				startByte = byteRange.from;
+				length = byteRange.to - byteRange.from + 1;
+				status = 206;
+
+				// Set Content-Range response header (we advertise initial resource size on server here (stat.size))
+				headers['Content-Range'] = `bytes ${byteRange.from}-${byteRange.to}/${stat.size}`;
+			} else {
+				byteRange.valid = false;
+				console.warn('Range request exceeds file boundaries, goes until byte no', byteRange.to, 'against file size of', length, 'bytes');
+			}
+		}
+
+		/* In any case, check for unhandled byte range headers */
+		if (!byteRange.valid && req.headers['range']) {
+			console.error(new Error('Range request present but invalid, might serve whole file instead'));
+		}
+
+		// Copy default headers
+		for (const k in this.options.headers) { headers[k] = this.options.headers[k]; }
+
+		headers['Etag'] = JSON.stringify([stat.ino, stat.size, mtime].join('-'));
+		headers['Date'] = new Date().toUTCString();
+		headers['Last-Modified'] = new Date(stat.mtime).toUTCString();
+		headers['Content-Type'] = contentType;
+		headers['Content-Length'] = length as any;
+
+		// Copy custom headers
+		for (const k in _headers) { headers[k] = _headers[k]; }
+
+		// Conditional GET
+		// If the "If-Modified-Since" or "If-None-Match" headers
+		// match the conditions, send a 304 Not Modified.
+		if ((clientMTime || clientETag) &&
+			(!clientETag || clientETag === headers['Etag']) &&
+			(!clientMTime || clientMTime >= mtime)) {
+			// 304 response should not contain entity headers
+			for (const entityHeader of [
+				'Content-Encoding', 'Content-Language', 'Content-Length', 'Content-Location', 'Content-MD5', 'Content-Range', 'Content-Type', 'Expires', 'Last-Modified',
+			]) {
+				delete headers[entityHeader];
+			}
+			return this.getResult(304, headers);
+		} else {
+			res.writeHead(status, headers);
+
+			if (req.method === 'HEAD') {
+				return this.getResult(status, headers, true);
+			}
```

Comment:
After sleeping and testing, this is how it should be, making proper use of the ``result -> finish`` flow. I also had forgotten to end the response. That will be consistently handled by ``finish()`` now.

```suggestion
		} else if (req.method === 'HEAD') {
			return this.getResult(status, headers);
		} else {
			res.writeHead(status, headers);
```